src/phonetics/french/sonnex.js

```diff
@@ -410,7 +410,7 @@ const RULES = {
   u: [
     ['un', '1'],
     ['ussent', 'us'],
-    [/^us(.*)/, (c, cs) => {
+    [/^us(.)(.*)/, (c, cs) => {
```

Comment:
Could you just add a comment over this line telling that the original algorithm was modified? This library is used as a reference point for some papers about phonetic algorithms and it would be cool to track what's original and what's not.

Reply:
> Could you just add a comment over this line telling that the original algorithm was modified? This library is used as a reference point for some papers about phonetic algorithms and it would be cool to track what's original and what's not.

By "this library" do you mean Sonnex or Talisman?
This fix aims to correct the regexp used in the current JavaScript implementation, which appears to use incorrect matching groups. As a consequence, for instance, `rébus` will produce the Sonnex `rEbuzundefine`, while `rEbu` is expected. On the other hand, the original Haskell code of Sonnex uses pattern-matching : https://github.com/Zigazou/Sonnex/blob/master/src/Text/Sonnex.hs#L348
This fix is inspired by the current handling of other cases such as `ti` or `om`. So I believe it does not really concern the Sonnex algorithm itself, but only this JavaScript implementation.

Reply:
I mean talisman.

> On the other hand, the original Haskell code of Sonnex uses pattern-matching

Yes. I don't remember the details but if the results of your fix produce the same results as the haskell version then no comment is needed indeed. I will trust you on this.

> This fix is inspired by the current handling of other cases such as ti or om.

Did I forget to comment them?

Reply:
> Yes. I don't remember the details but if the results of your fix produce the same results as the haskell version then no comment is needed indeed. I will trust you on this.

@Yomguithereal I've done a little testing with the Haskell lib and the results do seem consistent with the latest stable version of Sonnex, 0.1.0.3 (which is quite old actually... but the version on the `master` branch appears to be broken)

```
*Text.Sonnex> sonnex "usinage"
"uzinaj"
*Text.Sonnex> sonnex "écusson"
"Ekus3"
*Text.Sonnex> sonnex "intrus"
"1tru"
*Text.Sonnex> sonnex "rébus"
"rEbu"
*Text.Sonnex> sonnex "robuste"
"robust"
*Text.Sonnex> sonnex "plus"
"plu"
```
> > This fix is inspired by the current handling of other cases such as ti or om.
> 
> Did I forget to comment them?

What I mean is that these cases are already handled as expected, and I don't think any further comments are needed.